ibis/backends/duckdb/compiler.py

```diff
@@ -239,11 +239,8 @@ def visit_Cast(self, op, *, arg, to):
 
         return self.cast(arg, to)
 
-    @visit_node.register(ops.Literal)
-    def visit_Literal(self, op, *, value, dtype, **kw):
-        if value is None:
-            return super().visit_node(op, value=value, dtype=dtype, **kw)
-        elif dtype.is_interval():
+    def visit_NonNullLiteral(self, op, *, value, dtype):
```

Comment:
Can you pull out the duckdb, datafusion and clickhouse related changes to a different PR?

Reply:
Not really, because they are refactors of the base compiler.